server/graphql/v2/mutation/TransactionImportsMutations.ts

```diff
@@ -213,35 +221,52 @@ const transactionImportsMutations = {
       // Preload orders
       return sequelize.transaction(async transaction => {
         // Update rows
-        await Promise.all(
-          args.rows.map(async row => {
-            const rowId = idDecode(row.id, 'transactions-import-row');
-            const values = omitBy(omit(row, ['id', 'order']), isUndefined);
-            if (row.amount) {
-              values.amount = getValueInCentsFromAmountInput(row.amount);
-              values.currency = row.amount.currency;
-            }
-            if (row.order) {
-              const orderId = getDatabaseIdFromOrderReference(row.order);
-              const order = await req.loaders.Order.byId.load(orderId);
-              const collective = order && (await req.loaders.Collective.byId.load(order.CollectiveId));
-              if (!order || !collective || collective.HostCollectiveId !== transactionsImport.CollectiveId) {
-                throw new Unauthorized(`Order not found or not associated with the import: ${orderId}`);
+        if (args.rows?.length) {
+          await Promise.all(
+            args.rows.map(async row => {
+              const rowId = idDecode(row.id, 'transactions-import-row');
+              const values = omitBy(omit(row, ['id', 'order']), isUndefined);
```

Comment:
Ideally we create an array of editable fields and use `pick` here.

Reply:
It's not new, but I agree `pick` is safer than `omit`. Will fix that in a separate PR.